app/src/main/java/pm/gnosis/heimdall/ui/tokens/TokensFragment.kt

```diff
@@ -108,66 +102,33 @@ class TokensFragment : BaseFragment() {
                 .subscribeBy(onNext = this::showTokenRemovalDialog, onError = Timber::e)
     }
 
-    private fun onTokensList(tokens: List<ERC20Token>) {
-        adapter.setItems(tokens)
-        layout_tokens_empty_view.visibility = if (tokens.isEmpty()) View.VISIBLE else View.GONE
+    private fun onTokensList(tokens: Adapter.Data<TokensContract.ERC20TokenWithBalance>) {
+        adapter.updateData(tokens)
+        layout_tokens_empty_view.visibility = if (tokens.entries.isEmpty()) View.VISIBLE else View.GONE
     }
 
     private fun onTokensListError(throwable: Throwable) {
         Timber.e(throwable)
```

Comment:
see above (errorHandler might log it, maybe we should change the default behaviour to false for that)